sys/shell/cmds/gnrc_icmpv6_echo.c

```diff
@@ -207,7 +208,13 @@ static int _configure(int argc, char **argv, _ping_data_t *data)
                     /* intentionally falls through */
                 case 's':
                     if ((++i) < argc) {
-                        data->datalen = atoi(argv[i]);
+                        value = atoi(argv[i]);
+
+                        if (value < 0 || value > (int)(UINT16_MAX - sizeof(icmpv6_hdr_t))) {
```

Comment:
Value is checked for being negative here, so it is safer to cast `value` to `unsigned` here. Also some parens would be nice for better readability.

```suggestion
                        if ((value < 0) || ((unsigned)value > (UINT16_MAX - sizeof(icmpv6_hdr_t)))) {
```